### PR TITLE
[SPARK-41189][PYTHON] Add an environment to switch on and off namedtuple hack

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR is a followup of https://github.com/apache/spark/pull/34688 that adds a switch to turn on and off the namedtuple hack.

### Why are the changes needed?

There are still behaviour differences between regular pickle and Cloudpickle e.g., bug fixes from the upstream. It's safer to have a switch to turn on and off for the time being.

### Does this PR introduce _any_ user-facing change?

This remains as an internal environment so ideally no. In fact the main change itself was the internal change too.

### How was this patch tested?

Manually tested.

### CODE REVIEW

File: python/pyspark/serializers.py [hunk 1]
```diff
@@ -357,7 +358,7 @@ def dumps(self, obj):
         return obj
 
 
-if sys.version_info < (3, 8):
+if sys.version_info < (3, 8) or os.environ.get("PYSPARK_MONKEYPATCH_NAMEDTUPLE") == "1":
```
Comment (sadikovi) [inline python/pyspark/serializers.py:361]: nit: I was thinking about a name like this one: `PYSPARK_ENABLE_NAMEDTUPLE_PATCH` but I am open to alternatives.

**Review by sadikovi — APPROVED** [review]

File: python/pyspark/serializers.py [hunk 2]
```diff
@@ -54,6 +54,7 @@
 """
 
 import sys
+import os
```
[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply python/pyspark/serializers.py:57]: eh, it's actually fine in Python import (per PEP 8)

File: python/pyspark/serializers.py (hunk 1 above)
Comment (HyukjinKwon) [inline python/pyspark/serializers.py:361]:
```suggestion
if sys.version_info < (3, 8) or os.environ.get("PYSPARK_ENABLE_NAMEDTUPLE_PATCH") == "1":
```

File: python/pyspark/serializers.py [hunk 3]
```diff
@@ -471,7 +472,7 @@ def loads(self, obj, encoding="bytes"):
         return cloudpickle.loads(obj, encoding=encoding)
 
 
-if sys.version_info < (3, 8):
+if sys.version_info < (3, 8) or os.environ.get("PYSPARK_MONKEYPATCH_NAMEDTUPLE") == "1":
```
Comment (HyukjinKwon) [inline python/pyspark/serializers.py:475]:
```suggestion
if sys.version_info < (3, 8) or os.environ.get("PYSPARK_ENABLE_NAMEDTUPLE_PATCH") == "1":
```

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**